### PR TITLE
Replace module events

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- start a new model
- add a couple of apps
- relate them
- check that apps fade appropriately
- zoom in and out
- click on apps in the inspector and the canvas should center on them
- create and remove relations between apps
- resize the window, check that the apps get centered
- remove all apps, the onboarding should appear again
- play with the canvas as much as possible
- note: #1712 still exists

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7185/

File: jujugui/static/gui/src/app/views/topology/panzoom.js [hunk 1]
```diff
@@ -73,6 +73,24 @@ YUI.add('juju-topology-panzoom', function(Y) {
     },
 
     /**
+     * Handler for 'zoom_in' event.
+     *
+     * @method zoom_in
+     */
+    zoom_in: function(evt) {
+      // Not implemented.
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/views/topology/panzoom.js]: I've filed an issue.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7187/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7203/

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -303,7 +303,11 @@ YUI.add('juju-gui', function(Y) {
           if (spec.callback) { spec.callback.call(this, evt, target); }
           // HACK w/o context/view restriction but right direction
           if (spec.fire) {
-            this.views.environment.instance.topo.fire(spec.fire);
+            if (spec.fire.startsWith('topo.')) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: If the fire starts with topo then dispatch a new event else fire it from the topo? 

Can you add a comment to explain this, as it seems backwards :)

**Comment by hatched** [comment]
QA OK

Thanks @huwshimi.

The don't pan to application from a previous branch is a little odd when an application icon is mostly off screen and it still doesn't pan to it, but that's separate from this branch.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7226/

**Comment by huwshimi** [comment]
Thanks @Lukewh @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2300

**Comment by huwshimi** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**